tweet-wall-example/script.js

```diff
@@ -1,131 +1,129 @@
-async function getTweets() {
-    const response = await fetch(`${API_URL}/tweets`);
-    if (!response.ok) {
-        throw new Error('Failed to fetch tweets');
+// async function getCards() {
+//     const response = await fetch(API_URL);
+//     const data = await response.json();
+//     console.log(data);
+//     const deckID = data.deck_id;
+//     return deckID;
+// }
+
+// const deckID = getCards();
+class Card {
+    constructor(element, value, suit, image){
+        this.element = element; //creates the overall element
+        this.value = value; //creates the name 
+        this.suit = suit;
+        this.image = image; //creates gpa
+
+        this.element.classList.add('placedCard');
+        // console.log(this);
+    }
+
+    displayID (){
+        this.element.innerHTML = "";
+
+        const imageElement = document.createElement('img');
+        imageElement.classList.add('image');
+        imageElement.src = this.image;
+        imageElement.alt = 'image of the playing card';
+
+        const valueElement = document.createElement('p');
+        valueElement.classList.add('value');
+        valueElement.innerText = this.value + " of " + this.suit;
+
+        this.element.appendChild(imageElement);
+        this.element.appendChild(valueElement);
     }
-    const data = await response.json();
-    console.log(data);
-    return data;
 }
 
-async function getUser(id) {
-    const response = await fetch(`${API_URL}/users/${id}`);
-    if (!response.ok) {
-        throw new Error('Failed to fetch user');
+
+async function createDeck(){
+    contentElement.innerHTML = '';
+    const newResponse = await fetch(API_URL); //creates call
+    const newData = await newResponse.json(); //creates object that stores call
+    const deckID = newData.deck_id; //creates an object with the deck id
+    const deckURL = "https://deckofcardsapi.com/api/deck/" + deckID + "/draw/?count=5" //creates a url with new deck id
+   
+    const otherResponse = await fetch(deckURL); //fetches 3 cards from this deck
+    const newSet = await otherResponse.json(); //creates js object of the three cards
+    console.log(deckURL);
+    console.log(newSet); //logs in console
+   
+    const cardArray = newSet.cards; //short name
+        for (let i = 0; i < cardArray.length; i++){ //goes through every card in the array
+        const idEl = document.createElement('div');
+        contentElement.appendChild(idEl);
+        const newCard = new Card(idEl, cardArray[i].value, cardArray[i].suit, cardArray[i].image)
+        newCard.displayID();
     }
-    const data = await response.json();
-    console.log(data);
-    return data;
+
+    return deckURL;
 }
 
-async function renderTweet(tweet) {
-    const tweetEl = document.createElement('div');
-    tweetEl.classList.add('post-it');
+const deckURL = createDeck();
+console.log(deckURL)
+
+async function reDraw() {
+    const newResponse = await fetch(deckURL); //creates call
```

Comment:
Currently, `deckURL` is return a `Promise<>` because you didn't wait for `createDeck()` to finish before you tried to call `reDraw()`. If you are trying to call reDraw right after you call `createDeck()` should put `.then()` after `createDeck()` on line 71. Although, is that what you're actually intending to do? Or were you trying to call `reDraw()` automatically because you were trying to test it, since your click event wasn't working?